### PR TITLE
Fix for failing tests since new Firedrake update

### LINKED ISSUE #138: Tests fail
Newest test suite has failing tests (https://github.com/mesh-adaptation/movement/actions/runs/12221232036/job/34412642275)

```
=========================== short test summary info ============================
FAILED test/test_forced_movement.py::TestSpringMover::test_fixed_square - ValueError: ('on_boundary',) contains invalid boundary tags.
FAILED test/test_forced_movement.py::TestSpringMover::test_fixed_triangle - ValueError: ('on_boundary',) contains invalid boundary tags.
FAILED test/test_spring.py::TestExceptions::test_invalid_bc_tag_valueerror - AssertionError: '(4,) contains invalid boundary tags.' != '[4] contains invalid boundary tags.'
- (4,) contains invalid boundary tags.
? ^ ^^
+ [4] contains invalid boundary tags.
? ^ ^
FAILED test/test_spring.py::TestExceptions::test_tangling_valueerror - AssertionError: "('on_boundary',) contains invalid boundary tags." != 'Mesh has 1 tangled element.'
- ('on_boundary',) contains invalid boundary tags.
+ Mesh has 1 tangled element.
FAILED test/test_spring.py::TestStiffness::test_boundary_conditions_1x1_square_all_boundary - ValueError: ('on_boundary',) contains invalid boundary tags.
FAILED test/test_spring.py::TestStiffness::test_boundary_conditions_triangle_all_boundary - ValueError: ('on_boundary',) contains invalid boundary tags.
=========== 6 failed, 122 passed, 675 warnings in 769.77s (0:12:49) ============
```

### PR DESCRIPTION
**Pull request by ddundo**
Closes #138

Firedrake now returns `boundary_condition.sub_domain` as a tuple ( https://github.com/firedrakeproject/firedrake/pull/3905/files#diff-5dfb97c8200daf2aa6c38c106bcf5c6d85fde86e14a56f851cf6d03cdf91245b )

### CODE REVIEW

File: movement/spring.py [hunk 1]
```diff
@@ -205,9 +205,8 @@ def assemble_stiffness_matrix(self, boundary_conditions=None):
             # Determine boundary subsets for the associated tags
             bnd = self.mesh.exterior_facets
             tags = boundary_condition.sub_domain
-            if tags == "on_boundary":
+            if tags == ("on_boundary",):
                 tags = bnd.unique_markers
-            tags = [tags] if not isinstance(tags, Iterable) else tags
```
Comment (ddundo) [inline movement/spring.py:210]: Pretty sure `isinstance(tags, Iterable)` is now always `True`

**Review by jwallwork23 — CHANGES_REQUESTED** [review]
Thanks for this. The coverage is no longer at 100% (see https://github.com/mesh-adaptation/movement/pull/139/checks):
```
Name                              Stmts   Miss  Cover   Missing
---------------------------------------------------------------
movement/__init__.py                  4      0   100%
movement/laplacian_smoothing.py      47      0   100%
movement/math.py                     23      0   100%
movement/monge_ampere.py            270      2    99%   155, 166
movement/mover.py                   102      0   100%
movement/solver_parameters.py         7      0   100%
movement/spring.py                  164      0   100%
movement/tangling.py                 42      0   100%
---------------------------------------------------------------
TOTAL                               659      2    99%
```
Lines 154-155 of `monge_ampere` can be dropped similarly I think. L166 also looks related.

File: movement/spring.py (hunk 1 above)
Comment (jwallwork23) [reply movement/spring.py:210]: Yes I think you're correct.

**Comment by jwallwork23** [comment]
Oh actually no - L166 is fixed by https://github.com/mesh-adaptation/movement/pull/134. I hadn't realised that we hadn't merged that. Please could you review?

**Comment by ddundo** [comment]
> Thanks for this. The coverage is no longer at 100% (see https://github.com/mesh-adaptation/movement/pull/139/checks):
> Lines 154-155 of `monge_ampere` can be dropped similarly I think. L166 also looks related.

Sorry, I need to get into the habit of checking this!

> Oh actually no - L166 is fixed by #134. I hadn't realised that we hadn't merged that. Please could you review it?

I forgot about that again. I'm on it :)

**Comment by jwallwork23** [comment]
> Sorry, I need to get into the habit of checking this!

At some point I looked into setting up a code coverage check that would fail if the coverage decreased due to a PR but I couldn't find a decent solution for free. There's a tool called code-cov but I think you have to pay for it.

**Pull request merged**